### PR TITLE
[SPARK-23823][SQL] Keep origin in transformExpression

### PR DESCRIPTION
**Pull request by jiahuijiang**
Fixes https://issues.apache.org/jira/browse/SPARK-23823

Keep origin for all the methods using transformExpression

## What changes were proposed in this pull request?

Keep origin in transformExpression

## How was this patch tested?

Manually tested that this fixes https://issues.apache.org/jira/browse/SPARK-23823 and columns have correct origins after Analyzer.analyze

### CODE REVIEW

**Comment by gatorsmile** [comment]
ok to test

**Comment by cloud-fan** [comment]
LGTM, too

**Comment by JiahuiJiang** [comment]
@gatorsmile fixed :D

**Comment by hvanhovell** [comment]
Can you add a test? Just show that origin is still the same after you transform the node.

**Comment by JiahuiJiang** [comment]
@hvanhovell added a test, before this change the origin would be undefined

**Comment by cloud-fan** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/QueryPlanSuite.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.dsl.plans
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, Expression, Literal, NamedExpression}
+import org.apache.spark.sql.catalyst.trees.{CurrentOrigin, Origin}
+import org.apache.spark.sql.types.IntegerType
+
+class QueryPlanSuite extends SparkFunSuite {
+
+  test("origin remains the same after mapExpressions") {
```
Comment (kiszk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/QueryPlanSuite.scala]: nit: is it better to add JIRA number (i.e. `SPARK-PARK-23823: ...`)

**Comment by JiahuiJiang** [comment]
@kiszk Added the ticket number! Do we need to retrigger the test build on that new commit?

**Comment by JiahuiJiang** [comment]
Don't know why it failed so many times... but all these tests pass on my local machine and the commit before adding the test...

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merged to master/2.3

**Pull request closed without merging**